src/MainUpdate.elm

```diff
@@ -1053,6 +1102,95 @@ update msg model =
                 in
                     ( { model | displayedColumn = Just intColumnIdToDisplay }, Cmd.none )
 
+            FileRead data ->
+                let
+                    newFileId =
+                        Result.withDefault 0 (String.toInt data.id)
+
+                    newFile =
+                        { id = newFileId
+                        , contents = data.contents
+                        , filename = data.filename
+                        , filetitle = data.filetitle
+                        }
+
+                    newFilesToSave =
+                        List.filter (\f -> f.id /= newFileId) model.filesToSave
+                in
+                    ( { model | filesToSave = List.append newFilesToSave [ newFile ], showSavingFilesSpinner = False }
+                    , Cmd.none
+                    )
+
+            FileSelected id ->
+                ( { model | showSavingFilesSpinner = True }
+                , fileSelected (toString id)
```

Comment:
Having small integers like "1" as DOM ids is dangerous as they can easily be repeated elsewhere in the page and introduce error.